osm2streets/src/transform/separate_cycletracks.rs

```diff
@@ -101,11 +101,7 @@ fn snap(streets: &mut StreetNetwork, input: Cycleway) {
     assert!(streets.roads.contains_key(&input.cycleway));
 
     // Remove the cycleway, but remember the lanes it contained
-    let mut cycleway_lanes = streets
-        .roads
-        .remove(&input.cycleway)
-        .unwrap()
-        .lane_specs_ltr;
+    let mut cycleway_lanes = streets.remove_road(input.cycleway).lane_specs_ltr;
```

Comment:
This was a huge bug, because the two intersections' list of roads got out of sync!